pkg/exchange/coinbase/exchage.go

```diff
@@ -317,17 +319,20 @@ func (e *Exchange) QueryKLines(ctx context.Context, symbol string, interval type
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to get klines(%v): %v", interval, symbol)
 	}
-	if len(rawCandles) > options.Limit {
-		rawCandles = rawCandles[:options.Limit]
-	}
-	klines := make([]types.KLine, 0, len(rawCandles))
+	candles := make([]api.Candle, 0, len(rawCandles))
 	for _, rawCandle := range rawCandles {
 		candle, err := rawCandle.Candle()
 		if err != nil {
-			log.Warnf("invalid raw candle detected, skipped: %v", rawCandle)
+			log.Warnf("invalid raw candle detected, skipping: %v", rawCandle)
 			continue
 		}
-		klines = append(klines, toGlobalKline(symbol, granity, candle))
+		candles = append(candles, *candle)
+	}
+
+	klines := make([]types.KLine, 0, len(candles))
+	for _, candle := range candles {
+		kline := toGlobalKline(symbol, interval, &candle)
```

Comment:
Bbgo uses go 1.21.6. Before go 1.22, the variable between each for loop iteration is the same variable. That means the `&candle` always points to the same address unless you use go 1.22. Please ensure this behavior is expected.

https://go.dev/blog/loopvar-preview

Reply:
![截圖 2025-02-26 下午2 23 28](https://github.com/user-attachments/assets/dfc5cf28-65a8-41d6-8dd0-f2f36236dc7d)
Since I do not keep the pointer (`&candle`) in the body of `toGlobalKline()`, I think it will do what is expected in go 1.21.6.

Reply:
![截圖 2025-02-26 下午3 06 05](https://github.com/user-attachments/assets/1f3e8ce2-95da-41b6-9190-8087e859c346)
![截圖 2025-02-26 下午3 06 14](https://github.com/user-attachments/assets/f41f842c-4bb2-4050-8308-2c8c609e6e0b)

I compile and run it with go1.21.6
As shown in the screenshot, there should be no issue where pointers outlive the loop.